rusk-wallet/src/cache.rs

```diff
@@ -98,6 +101,9 @@ impl Cache {
             .map_err(|_| Error::Rkyv)?;
         let key = nullifier.to_bytes();
 
+        // keep track of max block height while inserting notes
+        self.insert_last_block_height(block_height)?;
+
         self.db.put_cf(&cf, key, data)?;
 
         Ok(())
```

Comment:
`insert` and `insert_spend` are the same methods with a slightly different `cf_name`.
Can you unify them to avoid code duplication?